init-interfaces.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -ex
+
+is_con_exists() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+is_con_active() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show --active | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+
+if [[ ! -f /boot/mac_addresses ]] ; then
+  echo "no mac address configuration file found .. exiting"
+  exit 1
+fi
+
+primary_mac=$(cat /boot/mac_addresses | awk -F= '/PRIMARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+secondary_mac=$(cat /boot/mac_addresses | awk -F= '/SECONDARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+
+default_device=""
+secondary_device=""
+default_connection_name=""
+secondary_connection_name=""
+
+for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+  dev_mac=$(nmcli -g GENERAL.HWADDR dev show $dev | sed -e 's/\\//g' | tr '[A-Z]' '[a-z]')
+  case $dev_mac in
+    $primary_mac)
+      default_device=$dev
+      default_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    $secondary_mac)
+      secondary_device=$dev
+      secondary_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    *)
+      ;;
+   esac
+done
+
+echo -e "default dev: $default_device ($default_connection_name)\nsecondary dev: $secondary_device ($secondary_connection_name)"
```

Comment:
Should we `exit 1` if the primary/secondary connection/device are empty strings?
I tend to yes, but wanted to get your input on that

Reply:
There can be no connections pre-defined for the interfaces and it would still be a healthy state. If https://github.com/RHsyseng/rhcos-slb/blob/main/custom-config.fcc#L18 worked as intended, that would be the case during the first boot.

Indeed if there are no devices, there is no point in continuing.

Reply:
yep. I'll `exit 1` to make sure the devices were collected